File: backend/src/main/java/biali/fitmanager/backend/dto/ProgressSummaryResponse.java
package biali.fitmanager.backend.dto;

import java.util.List;

public class ProgressSummaryResponse {
    private int daysSinceFirstTraining;
    private String dateRange;
    private List<ExerciseProgress> progressList;
    private List<Integer> chartData; // np. wartości do narysowania słupków

    // Wewnętrzna klasa reprezentująca wiersz na liście (np. Martwy ciąg: 60kg -> 100kg)
    public static class ExerciseProgress {
        private String exerciseName;
        private int startWeight;
        private int endWeight;

        public ExerciseProgress(String exerciseName, int startWeight, int endWeight) {
            this.exerciseName = exerciseName;
            this.startWeight = startWeight;
            this.endWeight = endWeight;
        }

        // Gettery i Settery
        public String getExerciseName() { return exerciseName; }
        public void setExerciseName(String exerciseName) { this.exerciseName = exerciseName; }
        public int getStartWeight() { return startWeight; }
        public void setStartWeight(int startWeight) { this.startWeight = startWeight; }
        public int getEndWeight() { return endWeight; }
        public void setEndWeight(int endWeight) { this.endWeight = endWeight; }
    }

    // Gettery i Settery dla głównej klasy
    public int getDaysSinceFirstTraining() { return daysSinceFirstTraining; }
    public void setDaysSinceFirstTraining(int daysSinceFirstTraining) { this.daysSinceFirstTraining = daysSinceFirstTraining; }
    
    public String getDateRange() { return dateRange; }
    public void setDateRange(String dateRange) { this.dateRange = dateRange; }
    
    public List<ExerciseProgress> getProgressList() { return progressList; }
    public void setProgressList(List<ExerciseProgress> progressList) { this.progressList = progressList; }
    
    public List<Integer> getChartData() { return chartData; }
    public void setChartData(List<Integer> chartData) { this.chartData = chartData; }
}

File: FitManager/app/src/main/java/biali/fitmanager/network/ProgressSummaryResponse.kt
package biali.fitmanager.network

import com.google.gson.annotations.SerializedName

data class ProgressSummaryResponse(
    @SerializedName("daysSinceFirstTraining")
    val daysSinceFirstTraining: Int,

    @SerializedName("dateRange")
    val dateRange: String,

    @SerializedName("progressList")
    val progressList: List<ExerciseProgress>,

    @SerializedName("chartData")
    val chartData: List<Int>
)

data class ExerciseProgress(
    @SerializedName("exerciseName")
    val exerciseName: String,

    @SerializedName("startWeight")
    val startWeight: Int,

    @SerializedName("endWeight")
    val endWeight: Int
)